File: tl/src/main/java/com/github/badoualy/telegram/tl/api/photos/TLAbsPhotos.java
package com.github.badoualy.telegram.tl.api.photos;

import com.github.badoualy.telegram.tl.api.TLAbsPhoto;
import com.github.badoualy.telegram.tl.api.TLAbsUser;
import com.github.badoualy.telegram.tl.core.TLObject;
import com.github.badoualy.telegram.tl.core.TLVector;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public abstract class TLAbsPhotos extends TLObject {
    protected TLVector<? extends TLAbsPhoto> photos;

    protected TLVector<? extends TLAbsUser> users;

    public TLAbsPhotos() {
    }

    public TLVector<? extends TLAbsPhoto> getPhotos() {
        return photos;
    }

    public void setPhotos(TLVector<? extends TLAbsPhoto> photos) {
        this.photos = photos;
    }

    public TLVector<? extends TLAbsUser> getUsers() {
        return users;
    }

    public void setUsers(TLVector<? extends TLAbsUser> users) {
        this.users = users;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/photos/TLAbsPhotos.kt
package com.github.badoualy.telegram.tl.api.photos

import com.github.badoualy.telegram.tl.api.TLAbsPhoto
import com.github.badoualy.telegram.tl.api.TLAbsUser
import com.github.badoualy.telegram.tl.core.TLObject
import com.github.badoualy.telegram.tl.core.TLObjectVector

/**
 * Abstraction level for the following constructors:
 * * [photos.photos#8dca6aa5][TLPhotos]
 * * [photos.photosSlice#15051f54][TLPhotosSlice]
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
abstract class TLAbsPhotos : TLObject() {
    abstract var photos: TLObjectVector<TLAbsPhoto>

    abstract var users: TLObjectVector<TLAbsUser>
}
